### ISSUE TITLE
Resource overrideMimeType not working on Edge

### ISSUE THREAD

**Issue by thw0rted** [AnalyticalGraphicsInc/cesium#7085]
TL;DR: I found a spec-compliance issue with how Resource does XHRs.

I came here to report an issue I was having with Edge, but to my surprise it isn't Edge's fault.  I have some code that makes a Resource and calls `fetchXML`, treating the response as an XMLDocument.  This failed in Edge, which led me to dig a bit -- I noticed that my server was returning the (definitely valid XML, not HTML) content with a `Content-Type` of `text/html`.  I don't know why, it's not under my control, so I'm stuck with it.  Firefox and Chrome return an XMLDocument anyway; Edge was returning an HTMLDocument.  I thought this was the bug, until I did a bunch of extra testing.

I found out that the XHR spec was [recently updated](https://github.com/whatwg/xhr/pull/174) to state that calling `open()` must reset the "override MIME type" -- see bullet 12 in [the current spec](https://xhr.spec.whatwg.org/#the-open()-method).  I saw that the Resource class [calls `overrideMimeType` before calling `open`](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Resource.js#L1860).  This results in the override getting ignored in recent builds of Edge, and the problem will get worse as other browsers come into compliance with the changes.

I *believe* the fix is as simple as moving the `open` call before the call to `overrideMimeType`.  This would be a PR instead of an issue, but my CLA issues remain unresolved, so I'll just point 👉

**Comment by thw0rted** [comment]
Update: WhatWG is [changing their mind](https://github.com/whatwg/xhr/pull/218) about this logic, so maybe the other browsers won't be matching Edge after all.  Still, the fact remains that this breaks on *some* shipping Edge versions and I don't think it hurts to reorder the logic in Resource.

**Comment by hpinkos** [comment]
Hmm, I wonder if this is related so some other weirdness we've been seeing in Edge recently.  Thanks for bring this up @thw0rted!

@mramato @tfili what are your thoughts on this?

**Comment by mramato** [comment]
I'm not sure this was the same thing I was seeing, but couldn't hurt.  Thanks for the write up @thw0rted, I'll open a PR just to cover our bases.

**Comment by mramato** [comment]
Thanks @thw0rted!  I was able to replicate this behavior and opened #7091 with the fix.

**Comment by hpinkos** [comment]
Thanks again for reporting this @thw0rted!  We just merged a fix for this into master and it will be included in the 1.50 release on Monday
